src/main/java/com/rcv/ContestConfig.java

```diff
@@ -107,59 +108,73 @@ private void validateCvrFileSources() {
     } else {
       HashSet<String> cvrFilePathSet = new HashSet<>();
       for (CVRSource source : rawConfig.cvrFileSources) {
+        // perform checks on source input path
         if (source.getFilePath() == null || source.getFilePath().isEmpty()) {
           isValid = false;
           Logger.log(Level.SEVERE, "filePath is required for each CVR file.");
-        } else if (cvrFilePathSet.contains(source.getFilePath())) {
+          continue;
+        }
+
+        // look for duplicate paths
+        if (cvrFilePathSet.contains(source.getFullFilePath())) {
           isValid = false;
           Logger.log(
-              Level.SEVERE, "Duplicate CVR filePaths are not allowed: %s", source.getFilePath());
+              Level.SEVERE, "Duplicate CVR filePaths are not allowed: %s",
+              source.getFullFilePath());
         } else {
-          cvrFilePathSet.add(source.getFilePath());
+          cvrFilePathSet.add(source.getFullFilePath());
         }
-
```

Comment:
Nit: for readability, it's nice to have a blank line between checks. Right now you randomly have them in some places and not in others in this method.